### DISCUSSION TITLE
pdk validate: Ability to include/exclude files & directories across all validators

### DISCUSSION THREAD

**Discussion by sanfrancrisko** [puppetlabs/pdk#1066] [Feature Request] [answered]
This is an attempt to clarify / define the requests across:
- [PDK-663](https://tickets.puppetlabs.com/browse/PDK-663)
- [PDK-1395](https://tickets.puppetlabs.com/browse/PDK-1395)
- #329 
- #341

The common request seems to have the ability to define an inclusion/exclusion list common to all validators, without having to configure each individual validator's config.

Exclusions for a specific validator should still be possible and the global config merged in to the more specific config.

We should define how and where we want to locate this global inclusion / exclusion list through this discussion

**Comment by cdenneen** [comment]
Has there been any traction on this. I have cloudformation templates that are failing the yaml validation. It just needs to be able to have exclusions.

What's interesting is:

```
pdk (ERROR): yaml-syntax: Unsupported class: Tried to load unspecified class: Date (cloudformation/apms_static.yaml)
```

```
❯ grep -i date cloudformation/apms_static.yaml
❯ grep -i class cloudformation/apms_static.yaml
❯ grep -i include cloudformation/apms_static.yaml
❯ grep -i load cloudformation/apms_static.yaml
❯
```

**Comment by jpogran** [comment]
We're adding this to the PDK 2.2.0 release planning, as this potentially requires changing the interface to validate.

We're going to release a 2.1.1 bugfix release before then, but wanted to update this discussion with the current progress.

**Comment by jpogran** [comment, marked as answer]
👋🏼 all! I have https://github.com/puppetlabs/pdk/pull/1114 up, which adds an ignore file list to `pdk validate`, for all of the validators. Its added as a project setting for pdk, which means you set it via `pdk set config`. This means the list of excluded files is now contained inside the project, so any future calls to pdk validate respects the list (for example in CI). 

While I work out adding the spec tests, I'd appreciate commentary on the UI of adding the setting and commiting that file to project.
